packages/plugin-data/src/commands/force/data/soql/query.ts

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as os from 'os';
+import { flags, FlagsConfig, SfdxCommand, SfdxResult } from '@salesforce/command';
+import { Connection, Logger, Messages, Org } from '@salesforce/core';
+import {
+  ensureAnyJson,
+  ensureJsonArray,
+  ensureJsonMap,
+  ensureString,
+  isJsonArray,
+  toJsonMap,
+} from '@salesforce/ts-types';
+import { Tooling } from '@salesforce/core/lib/connection';
+import { CsvReporter, FormatTypes, HumanReporter, JsonReporter } from '../../../../reporters';
+import { Field, FieldType, SoqlQueryResult } from '../../../../dataSoqlQueryTypes';
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages('@salesforce/plugin-data', 'soql.query');
+
+/**
+ * Class to handle a soql query
+ *
+ * Will collect all records and the column metadata of the query
+ */
+export class SoqlQuery {
+  public async runSoqlQuery(connection: Connection | Tooling, query: string, logger: Logger): Promise<SoqlQueryResult> {
+    let columns: Field[] = [];
+    logger.debug('running query');
+
+    const result = await connection.autoFetchQuery(query, { autoFetch: true, maxFetch: 50000 });
```

Comment:
@amphro, I am using the default maxFetch of 50000. This is the only way I could get on par with old query command that used query/queryMore up to 5000 records.

I learned today that there is a config setting `maxQueryLimit`. Should I be considering that setting and if not present, default to 50000 (or maybe 10000, which is a published default)?

Reply:
Can we just call `autoFetchQuery` with no options and leave everything to be the default? That warning message about the records returned is possibly a bug in core around `count()`. FYI @WillieRuemmele might be able to provide more info.

Reply:
@amphro, the default for maxFetch is 10000 records and maxQueryLimit is considered in autoFetch impl, however original behavior was query/queryMore until records were exhausted, delegating how many rows were returned to the query itself, i.e. `select id from Account LIMIT 1` or `select id from Account`, or `select id from Account LIMIT 49999`. Given 50000 records is the hard limit for a soql query, it seemed like the reasonable thing to do, while still allowing user to control records returned with `LIMIT` in queries.

So, removing maxFetch from soql query may change existing behavior. We would need to let customers know that in a release note.

Reply:
I see. I didn't understand how it worked before then. I would prefer to keep the behavior how it was before unless we have a good reason (performance, security, etc).

Reply:
@amphro It is still in its original form,`maxFetch: 50000`